### PR TITLE
Return better NWC error code for payment failures

### PR DESCRIPTION
**Pull request by benthecarman**

### CODE REVIEW

File: mutiny-core/src/nostr/nwc.rs [hunk 1]
```diff
@@ -1038,10 +1038,16 @@ impl NostrWalletConnect {
                                     }
                                 }
 
+                                let code = if e == MutinyError::InsufficientBalance {
+                                    ErrorCode::InsufficientBalance
+                                } else {
+                                    ErrorCode::PaymentFailed
```
Comment (TonyGiorgio) [inline mutiny-core/src/nostr/nwc.rs:1044]: payment timeout shouldn't give a failure

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

File: mutiny-core/src/nostr/nwc.rs (hunk 1 above)
Comment (benthecarman) [reply mutiny-core/src/nostr/nwc.rs:1049]: added

File: mutiny-core/src/nostr/nwc.rs [hunk 2]
```diff
@@ -999,6 +999,11 @@ impl NostrWalletConnect {
                                             nostr_manager.logger,
                                             "Payment timeout, not removing payment from budget"
                                         );
+
+                                        nostr_manager.save_nwc_profile(self.clone())?;
+
+                                        // don't save to pending list, we already paid it
+                                        return Ok(None);
```
Comment (TonyGiorgio) [inline mutiny-core/src/nostr/nwc.rs:1006]: Payments that timeout, but then fail, will not be included in the list for them to retry. That's a huge miss.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

File: mutiny-core/src/nostr/nwc.rs (hunk 2 above)
Comment (benthecarman) [reply mutiny-core/src/nostr/nwc.rs]: good catch

**Review by TonyGiorgio — APPROVED** [review]

**Pull request merged**